### PR TITLE
Move meta-properties about mutable refs to its spec module.

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: example/test/MutableReferenceSpec.hs [hunk 1]
```diff
@@ -1,9 +1,180 @@
...
+      all (\(Ex _ ref) -> ref `elem` refs) (uses' c) &&
+      go refs cs
+
+scopeCheckFork'
+  :: forall
+     (ix   :: *)
+     (cmd  :: Response ix -> (TyFun ix * -> *) -> *)
+  .  (forall resp. cmd resp ConstIntRef -> SResponse ix resp)
+  -> (forall resp. cmd resp ConstIntRef -> [Ex ConstIntRef])
+  -> Fork [IntRefed cmd] -> Bool
```
Comment (Danten) [inline example/test/MutableReferenceSpec.hs]: This doesn't mention `MemStep` so maybe it should be in the library somewhere?

File: example/test/MutableReferenceSpec.hs [hunk 2]
```diff
@@ -1,9 +1,180 @@
...
+      all (\(Ex _ ref) -> ref `elem` refs) (uses' c) &&
+      go refs' cs
+    SResponse     ->
+      all (\(Ex _ ref) -> ref `elem` refs) (uses' c) &&
+      go refs cs
+
+scopeCheckFork'
+  :: forall
+     (ix   :: *)
+     (cmd  :: Response ix -> (TyFun ix * -> *) -> *)
```
Comment (Danten) [inline example/test/MutableReferenceSpec.hs]: Don't need to provide the kind signatures.

File: example/test/MutableReferenceSpec.hs [hunk 3]
```diff
@@ -1,9 +1,180 @@
...
+  .  (forall resp. cmd resp ConstIntRef -> SResponse ix resp)
+  -> (forall resp. cmd resp ConstIntRef -> [Ex ConstIntRef])
```
Comment (Danten) [inline example/test/MutableReferenceSpec.hs]: Isn't this just `itoList` for `IxFoldable`?

File: example/test/MutableReferenceSpec.hs [hunk 4]
```diff
@@ -1,9 +1,180 @@
...
 import           MutableReference
+import           Test.StateMachine.Internal.Parallel
+import           Test.StateMachine.Internal.Sequential
+import           Test.StateMachine.Types
+import           Test.StateMachine.Types.AlphaEquality
+import           Test.StateMachine.Utils
+
+------------------------------------------------------------------------
+
+usesRefs :: MemStep resp refs -> [Ex refs]
```
Comment (Danten) [inline example/test/MutableReferenceSpec.hs]:
```
usesRefs = itoList
```
?

Comment (stevana) [inline example/test/MutableReferenceSpec.hs]: Jo!

File: example/test/MutableReferenceSpec.hs (hunk 2 above)
Comment (stevana) [inline example/test/MutableReferenceSpec.hs]: Fixed, thanks!

File: example/test/MutableReferenceSpec.hs (hunk 3 above)
Comment (stevana) [inline example/test/MutableReferenceSpec.hs]: Fixed, thanks!

File: example/test/MutableReferenceSpec.hs (hunk 1 above)
Comment (stevana) [inline example/test/MutableReferenceSpec.hs]: Agreed, but where? `.Types.ScopeCheck` (next to `.Types.AlphaEquality`)?

**Pull request merged**